derive/src/trait_bounds.rs

```diff
@@ -149,29 +160,49 @@ pub fn add(
 			.for_each(|ty| {
 				where_clause.predicates.push(parse_quote!(#ty : #has_compact_bound))
 			});
+
+		skip_types
+			.into_iter()
+			.for_each(|ty| {
+				let codec_skip_bound = codec_skip_bound.as_ref().unwrap();
+				where_clause.predicates.push(parse_quote!(#ty : #codec_skip_bound))
+			});
 	}
 
 	Ok(())
 }
 
-fn needs_codec_bound(field: &syn::Field) -> bool {
-	!crate::utils::get_enable_compact(field)
+fn needs_codec_bound(field: &syn::Field, variant_skip: bool) -> bool {
+	!variant_skip
+		&& !crate::utils::get_enable_compact(field)
 		&& crate::utils::get_encoded_as_type(field).is_none()
+		&& crate::utils::get_skip(&field.attrs).is_none()
 }
 
-fn needs_has_compact_bound(field: &syn::Field) -> bool {
-	crate::utils::get_enable_compact(field)
+fn needs_has_compact_bound(field: &syn::Field, variant_skip: bool) -> bool {
+	!variant_skip
+		&& crate::utils::get_enable_compact(field)
 }
 
-fn collect_types(data: &syn::Data, type_filter: fn(&syn::Field) -> bool) -> syn::Result<Vec<syn::Type>> {
+fn needs_default_bound(field: &syn::Field, variant_skip: bool) -> bool {
+	// types from skipped variants doesn't need default
+	!variant_skip
```

Comment:
I don't understand why each function takes `variant_skip` and does the same with it? Can not just `collect_types` do this?

Reply:
Yes true